extensions/auth/openldap/openldap.go

```diff
@@ -0,0 +1,117 @@
+package openldap
+
+import (
+	"fmt"
+
+	management "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+)
+
+type OLDAPOperations interface {
+	Enable() error
+	Disable() error
+}
+
+const (
+	resourceType = "openldap"
+	schemaType   = "openLdapConfigs"
+)
+
+type OLDAP struct {
+	Config *Config
+
+	session *session.Session
+	client  *rancher.Client
+}
+
+// NewOLDAP constructs OLDAP struct after it reads Open LDAP from the configuration file
+func NewOLDAP(client *rancher.Client, ts *session.Session) (*OLDAP, error) {
+	ldapConfig := new(Config)
+	config.LoadConfig(ConfigurationFileKey, ldapConfig)
+
+	return &OLDAP{
+		Config:  ldapConfig,
+		session: ts,
+		client:  client,
+	}, nil
+}
+
+// Enable is a method function of OLDAP, makes a request to the action with the given
+// configuration values
+func (o *OLDAP) Enable() error {
+	var jsonResp map[string]interface{}
```

Comment:
what is the purpose of this empty map? we are not using it anywhere other than the post request.. can we not just send an empty map instead of declaring a variable for it ?